WIP on Issue 21

BhanuKandula5030 opened the pull request:
This Issue 21 needs to register an endpoint pattern for /line which handles the GET method
Required URL parameters : /line and /line/{id} 

Changes Introduced:

1.Added a new module line.mjs to handle the /line endpoint with the specified requirements.
2.Modified app.mjs to include the new lineRouter for handling /line requests.
3.Updated existing code for better error handling and asynchronous behavior in the line-related functionality.
Testing

Reviewer  Steps experience that impact of your changes as part of this software repository:

1.Navigate to the Tpen services folder.
2.npm install (For installing the dependencies )
3..npm start (Running the application)
4.Terminal Response : Listening on 3001

Testing(Browser/Postman):

1.URL test : GET : http://localhost:3001/line/123
2.Test without the id  : GET : http://localhost:3001/line/
3.Test with id as not number   : GET: http://localhost:3001/page/apple
4.Test with not valid id number : GET :  http://localhost:3001/page/329

thehabes reviewed (changes requested):
Please only take PRs out of Draft when you feel you have completed all the requirements of your issue.  When you take a PR out of Draft that signals to us you have completed the requirements of your issue are having us review and test the code for submission and deployment.

There are no tests written.  Please write tests for your endpoint.

I was able to run the code and test it
http://localhost:3001/line does not have a good message
http://localhost:3001/line/potato does not have a good message

cubap commented on Code Changes Documnetation.md:
```diff
@@ -0,0 +1,74 @@
+# Code Documentation for the Changes 
```
and check the spelling of the filename ;)

BhanuKandula5030 replied:
Thank you for the feedback! I appreciate your suggestion . I'll make sure to follow this pattern and include relevant documentation . Regarding the spelling of the filename, I'll ensure to correct that in the next commit.Additionally, We will create a separate issue for "API documentation" where I'll document all the updates and changes. This will help us keep track of the documentation and make it easier for reference.

BhanuKandula5030 commented on line/line.mjs:
```diff
@@ -0,0 +1,100 @@
+    utils.respondWithError(res, 405, 'Improper request method, please use GET.');
+  });
+
+export async function findLineById(id = null) {
```
Thanks for the comment @thehabes I have made the separate files to handle the routing and logic

BhanuKandula5030 commented:
I have updated the code with the requested comments and added the end to end functionality,exists and functionality unit test cases and everything gets passed.

thehabes commented on README.md:
```diff
@@ -1,2 +0,0 @@
-# TPEN Services
```
I think you deleted this on accident.  We want this content as the stub for the README for now.

BhanuKandula5030 replied:
Thanks for letting me know @thehabes .I have added in the updated code.

thehabes commented on line/__tests__/functionality_unit.test.mjs:
```diff
@@ -0,0 +1,21 @@
+import app from '../../app.mjs';
```
You do not need this import.  I had thought we needed it for coverage previously, but after looking into it we don't need this import in the test files unless we use `app` in the test.

thehabes commented on line/__tests__/end_to_end_unit.test.mjs:
```diff
@@ -0,0 +1,58 @@
+
+  it('Call to /line with a TPEN 3 line "${id}" does not exist. The status should be 404 with a message.', async () => {
+    try {
+      const res = await request(routeTester).get('/line/1257');
```
Clean up extraneous `;`

thehabes commented on utilities/shared.mjs:
```diff
@@ -32,6 +32,18 @@ export function validateProjectID(id){
    return false
 }
 
+export function validateLineID(id) {
```
I think all functions will share a `validateID()` function.  This does the same thing as the validateProjectID() function, so we won't need it.  

I will clean this upstream after we merge.

BhanuKandula5030 replied:
Can we use the ProjectID here @thehabes ?

BhanuKandula5030 replied:
I agree with you and updated the same in the code .

thehabes reviewed (changes requested):
Just bring back the README content and this is good to go.
